src/main/java/core/basesyntax/Figure.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax;
+
+import core.basesyntax.enums.Color;
+import core.basesyntax.enums.FigureName;
+import core.basesyntax.interfaces.Area;
+import core.basesyntax.interfaces.Draw;
+
+public abstract class Figure implements Area, Draw {
+    private Color color = new ColorSupplier().getRandomColor();
```

Comment:
```suggestion
    private Color color;
```

Reply:
It is necessary not to set the color here, but to pass it through the constructor during creation.